### ISSUE TITLE
Teradata: Missing parsing [NOT] CASESPECIFIC for strings

### ISSUE THREAD

**Issue by rafaelkrysciak** [sqlfluff/sqlfluff#5530]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### Description

In Teradata, strings can be marked as `CASESPECIFIC` or `NOT CASESPECIFIC` by suffixing it with `(CASESPECIFIC)` or `(NOT CASESPECIFIC)`. There is also a short form of it `(CS)` or `(NOT CS)`.
Example:
```sql
SELECT
    CASE
        WHEN 'TEST' (CASESPECIFIC) = 'test' (CASESPECIFIC)
            THEN 'Not true'
        WHEN 'TEST' (NOT CASESPECIFIC) = 'test' (NOT CASESPECIFIC)
            THEN 'True'
        WHEN 'TEST' (CS) = 'test' (CS)
            THEN 'Not true'
        WHEN 'TEST' (NOT CS) = 'test' (NOT CS)
            THEN 'True'
    END
```
This syntax is not recognized yet. 

There are also side effects when the line is marked with `--noqa`. The line after got wrong parsed. Other errors are reported. For example, a missing entity in the FROM section. Ever the entity is definitely there. 

### Use case

There should be no error reported for this query:
```sql
SELECT
    CASE
        WHEN SOME_TABLE.Attribute1 = 'test' (CASESPECIFIC)
            THEN 'Not true'
        WHEN SOME_TABLE.Attribute1 = 'Test' (NOT CASESPECIFIC)
            THEN 'True'
    END
FROM SOME_DATASE.SOME_TABLE AS SOME_TABLE
```

### Dialect

Teradata

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
